### PR TITLE
feat: support zip for output artifacts archive. Fixes #8861

### LINKED ISSUE #8861: Support archive (none and zip) in output artifacts.
# Summary

The output archive format is tar.gz

# Use Cases

When zip or none archive are needed.

---

**Message from the maintainers**:

Love this enhancement proposal? Give it a 👍. We prioritise the proposals with the most 👍.

### PR DESCRIPTION
**Pull request by Belyenochi**
Support [#8861 ](https://github.com/argoproj/argo-workflows/issues/8861)

### CODE REVIEW

File: util/archive/archive_test.go
Comment (tczhao) [inline util/archive/archive_test.go:166]: .zip

Zipping

**Review by tczhao — CHANGES_REQUESTED** [review]
you can fix the [DCO](https://github.com/argoproj/argo-workflows/pull/8973/checks?check_run_id=6904105747) by
```
git commit -s --amend
git push -f
```

File: workflow/executor/executor_test.go
Comment (Belyenochi) [reply workflow/executor/executor_test.go:386]: Thanks for review, I'm not sure if I want to append a single test here, please give me some advice.

Comment (Belyenochi) [reply workflow/executor/executor_test.go]: > because `SaveArtifacts` actually deletes the artifact after saving, maybe you can wrap this whole test into a forloop, and use different archive strategy for each for loop element e.g. https://github.com/argoproj/argo-workflows/blob/master/persist/sqldb/archived_workflow_labels_test.go#L31

Thanks for your suggestion, I will resubmit later

**Review by tczhao — APPROVED** [review]

**Comment by Belyenochi** [comment]
@tczhao hi tzzhao, could you help me to check the error of the e2e test, I am more puzzled that I changed the executor, whether the error of this cli_test is a random error.

**Comment by tczhao** [comment]
Hi @Belyenochi 
Can you retrigger the CI by commit an empty commit
e.g. `git commit --allow-empty -sm "fix: empty" && git push`

I don't think the failed test is related to your change, maybe the test is flaky
You can try test the failed e2e test locally by
https://argoproj.github.io/argo-workflows/running-locally/#running-e2e-tests-locally

**Comment by Belyenochi** [comment]
> Hi @Belyenochi Can you retrigger the CI by commit an empty commit e.g. `git commit --allow-empty -sm "fix: empty" && git push`
> 
> I don't think the failed test is related to your change, maybe the test is flaky You can try test the failed e2e test locally by https://argoproj.github.io/argo-workflows/running-locally/#running-e2e-tests-locally

Resubmitted and please merge.

File: workflow/executor/executor.go
Comment (juliev0) [inline workflow/executor/executor.go:496]: Since this code seems to be a duplicate of above, I wonder if we should break it out into its own method or function?

**Review by juliev0 — COMMENTED** [review]
looks good. Added some minor comments.

**Comment by Belyenochi** [comment]
> looks good. Added some minor comments.

Thanks for review, I will fix it soon.

File: workflow/executor/executor.go
Comment (Belyenochi) [reply workflow/executor/executor.go:496]: I'm a little confused if I pull out this part of the code and bring it to the front, the code style may appear inconsistent, please give me some suggestions for merging.

Comment (Belyenochi) [reply workflow/executor/executor.go:496]: > Since this code seems to be a duplicate of above, I wonder if we should break it out into its own method or function?

I'm a little confused if I pull out this part of the code and bring it to the front, the code style may appear inconsistent, please give me some suggestions for merging.

Comment (juliev0) [reply workflow/executor/executor.go:496]: Okay, let me look at it.

Comment (juliev0) [reply workflow/executor/executor.go:496]: I know what you mean, that the tgz code isn't broken into a separate function. I suppose it is only about 10 lines that are duplicated. If the approver's okay with it, it's okay with me. Thanks for your patience.

**Review by juliev0 — APPROVED** [review]
There was some copy/pasting that I thought could be addressed, but it was <10 lines so thought it might be okay...will leave that up to the person who merges this.

**Comment by Belyenochi** [comment]
@sarabala1979 hi sarabala, Is there anything else I need to do here?

File: workflow/executor/executor.go
Comment (sarabala1979) [reply workflow/executor/executor.go:496]: Sorry for the delayed reply. I would prefer the separate function so if any bug is in this code, a developer can fix it in one place.
@juliev0 can you create techdept issue for other types?

Comment (juliev0) [reply workflow/executor/executor.go:496]: No problem. Once @Belyenochi makes his changes, I'll see what's left and create an issue for that. @Belyenochi if you have specific areas of concern regarding inconsistency let me know.

Comment (Belyenochi) [reply workflow/executor/executor.go:496]: @juliev0 I see, I'll try to decompose into a function.

Comment (juliev0) [reply workflow/executor/executor.go:496]: thanks!

**Review by alexec — APPROVED** [review]

**Pull request merged**
